hedera-node/src/main/java/com/hedera/services/sigs/sourcing/SigMapPubKeyToSigBytes.java

```diff
@@ -53,15 +50,18 @@
 
 	@Override
 	public byte[] sigBytesFor(byte[] pubKey) throws KeyPrefixMismatchException {
-		List<byte[]> matchingSigs = sigMap.getSigPairList()
-				.stream()
-				.filter(sp -> beginsWith(pubKey, sp.getPubKeyPrefix().toByteArray()))
-				.map(this::sigBytesFor)
-				.collect(toList());
-		if (matchingSigs.size() > 1) {
-			throw new KeyPrefixMismatchException("Source signature map is ambiguous for given public key!");
+		byte[] sigBytes = EMPTY_SIG;
+		for (SignaturePair sigPair : sigMap.getSigPairList()) {
+			final byte[] pubKeyPrefix = sigPair.getPubKeyPrefix().toByteArray();
+			final int length = pubKeyPrefix.length;
+			if (Arrays.equals(pubKeyPrefix, 0, length, pubKey, 0, length)) {
+				if (sigBytes != EMPTY_SIG) {
+					throw new KeyPrefixMismatchException("Source signature map is ambiguous for given public key!");
```

Comment:
For this `throw`, also add the info of `pubKeyPrefix` and `pubKey`. Need a test to cover this throw with the expected info.

Reply:
There is at least one unit test where we create an intentionally ambiguous `sigMap` and verify the  `KeyPrefixMismatchException` is thrown, here: [SigVerifierRegressionTest.throwsOnInvalidSigMap()](https://github.com/hashgraph/hedera-services/blob/master/hedera-node/src/test/java/com/hedera/services/sigs/SigVerifierRegressionTest.java#L187). 

Apparently it was unstable in CI because it has,
```
		assumeFalse(isInCircleCi);
```
But @Neeharika-Sompalli you could try removing the `assumeFalse(isInCircleCI)` from the `SigVerifierRegressionTest`...at some point I was to start running the `SmartContract*Test` unit tests in CircleCI for no obvious reason, maybe this can too now! 😁

Reply:
@tinker-michaelj  Sure I will uncomment the line. `SigMapPubKeyToSigBytesTest.rejectsNonUniqueSigBytes` also checks for the `KeyPrefixMismatchException`